### PR TITLE
Add simplified mappings "UploadClickConversion" action

### PR DESCRIPTION
**Pull request by marinhero**
This PR adds the simplified mappings version of the UploadClickConversion action. The action implements the Sync Mode for Simplified Mappings.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+  handleGoogleErrors,
+  convertTimestamp,
+  getApiVersion,
+  commonHashedEmailValidation,
+  getConversionActionDynamicData,
+  isHashedInformation
+} from '../functions'
+
+const action: ActionDefinition<Settings, Payload> = {
+  title: 'Upload Click Conversion (Simplified)',
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts:29]: Note the `(Simplified)` version.

File: packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+  description: 'Upload an offline click conversion to the Google Ads API.',
+  syncMode: {
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts:31]: SyncMode present

File: packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+  dynamicFields: {
+    conversion_action: async (
+      request: RequestClient,
+      { settings, auth, features, statsContext }
+    ): Promise<DynamicFieldResponse> => {
+      return getConversionActionDynamicData(request, settings, auth, features, statsContext)
+    }
+  },
+  perform: async (request, { auth, settings, payload, features, statsContext, syncMode }) => {
+    if (syncMode === 'add') {
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts:239]: only operates for this sync mode

File: packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts (hunk 2 above)
Comment (nick-Ag) [reply packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts:32]: Does google not support deleting conversions?

Comment (marinhero) [reply packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts:32]: Not according to their docs here -> https://developers.google.com/google-ads/api/docs/conversions/upload-clicks

File: packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts (hunk 3 above)
Comment (maryamsharif) [reply packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts:240]: Just to clarify this action will only be used with RETL?

**Review by maryamsharif — DISMISSED** [review]
Just left 1 non-blocking comment.

File: packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts (hunk 3 above)
Comment (marinhero) [reply packages/destination-actions/src/destinations/google-enhanced-conversions/uploadClickConversion2/index.ts:240]: For now, yes. We will bring sync mode into the regular UI after or default to a value.

**Review by nick-Ag — APPROVED** [review]

**Pull request merged**